### PR TITLE
[NO SQUASH] async env / mapgen fixes

### PR DESCRIPTION
**Pull request by sfan5**
some quick commits extracted from my other PR again

## To do

This PR is Ready for Review.

### CODE REVIEW

File: builtin/async/game.lua [hunk 1]
```diff
@@ -36,11 +36,16 @@ do
...
+			setmetatable(v, {__index = all.nodedef_default})
 			all.registered_nodes[k] = v
 		elseif v.type == "craftitem" then
+			setmetatable(v, {__index = all.craftitemdef_default})
 			all.registered_craftitems[k] = v
 		elseif v.type == "tool" then
+			setmetatable(v, {__index = all.tooldef_default})
 			all.registered_tools[k] = v
+		else
+			setmetatable(v, {__index = all.noneitemdef_default})
```
[the remark this answers is not in the document]
Comment (rubenwardy) [reply builtin/async/game.lua:48]: No, this sets/replaces the metatable - it doesn't add to it. You could do `getmetatable(v).__index = all.noneitemdef_default` although iirc getmetatable isn't whitelisted in the sandbox

**Review by rubenwardy — APPROVED** [review]
Looks good

File: src/mapgen/mapgen.cpp [hunk 2]
```diff
@@ -370,19 +370,25 @@ inline bool Mapgen::isLiquidHorizontallyFlowable(u32 vi, v3s16 em)
...
 	const v3s16 &em  = vm->m_area.getExtent();
 
+	isignored = wasignored = true;
+	isliquid = wasliquid = false;
+	was_n = CONTENT_IGNORE;
+
 	for (s16 z = nmin.Z + 1; z <= nmax.Z - 1; z++)
 	for (s16 x = nmin.X + 1; x <= nmax.X - 1; x++) {
-		wasignored = true;
-		wasliquid = false;
```
Comment (Desour) [inline src/mapgen/mapgen.cpp:378]: Why don't you have to reset these at each node column?

**Review by Desour — COMMENTED** [review]
Looks fine otherwise. :+1:

File: src/mapgen/mapgen.cpp (hunk 2 above)
Comment (sfan5) [reply src/mapgen/mapgen.cpp:378]: good point, I do

**Pull request merged**